nlp-plugins/src/main/java/io/cdap/google/plugins/NLPConfig.java

```diff
@@ -0,0 +1,154 @@
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.validation.InvalidConfigPropertyException;
+import io.cdap.google.common.NLPMethod;
+
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+
+/**
+ * A config for {@link NLPTransform} plugin
+ */
+public class NLPConfig extends PluginConfig {
+  public static final String AUTO_DETECT = "auto-detect";
+
+  public static final String PROPERTY_SOURCE_FIELD = "sourceField";
+  public static final String PROPERTY_METHOD_NAME = "methodName";
+  public static final String PROPERTY_ENCODING = "encoding";
+  public static final String PROPERTY_LANGUAGE_CODE = "languageCode";
+  public static final String PROPERTY_ERROR_HANDLING = "errorHandling";
+  public static final String PROPERTY_SERVICE_ACCOUNT_FILE_PATH = "serviceFilePath";
+
+  @Name(PROPERTY_SOURCE_FIELD)
+  @Description("Field which contains an input text")
+  @Macro
+  private String sourceField;
+
+  @Name(PROPERTY_METHOD_NAME)
+  @Description("Name of Google Natural Language API Method")
+  @Macro
+  private String methodName;
+
+  @Name(PROPERTY_ENCODING)
+  @Description("Text encoding. Providing it is recommended because the API provides the beginning offsets for" +
+    "various outputs, such as tokens and mentions, and languages that natively use different text encodings may" +
+    "access offsets differently.")
+  @Macro
+  @Nullable
+  private String encoding;
+
+
+  @Name(PROPERTY_LANGUAGE_CODE)
+  @Description("Code of the language of the text data. E.g. en, jp, etc. If not provided" +
+    "Google Natural Language API will autodetect the language.")
+  @Macro
+  @Nullable
+  private String languageCode;
+
+  @Name(PROPERTY_ERROR_HANDLING)
+  @Description("Error handling strategy to use when there is an during NLP API call.")
+  private String errorHandling;
+
+  @Name(PROPERTY_SERVICE_ACCOUNT_FILE_PATH)
+  @Description("Path on the local file system of the service account key used "
+    + "for authorization. Can be set to 'auto-detect' when running on a Dataproc cluster. "
+    + "When running on other clusters, the file must be present on every node in the cluster.")
+  @Macro
+  @Nullable
+  protected String serviceFilePath;
+
+  public String getSourceField() {
+    return sourceField;
+  }
+
+  public NLPMethod getMethod() {
+    return Stream.of(NLPMethod.class.getEnumConstants())
+      .filter(keyType -> keyType.getValue().equalsIgnoreCase(methodName))
+      .findAny()
+      .orElseThrow(() -> new InvalidConfigPropertyException(
+        String.format("Unsupported value for '%s': '%s'", PROPERTY_METHOD_NAME, methodName), PROPERTY_METHOD_NAME));
+  }
+
+  public EncodingType getEncodingType() {
+    if (encoding == null) {
+      return EncodingType.NONE;
+    }
+
+    switch (encoding) {
```

Comment:
why do you use switch here instead of stream like in method below?

Reply:
done